src/config/field-overrides/predictor/ExpressionField.js

```diff
@@ -56,11 +69,12 @@ class ExpressionField extends Component {
                 />
                 {props.errorText ? <div style={styles.errorText}>{props.errorText}</div> : null}
                 <ExpressionDialog
-                    {...props}
+                    {...expressionDialogProps}
```

Comment:
So this component is doing the following:
- In `handleOpen` the `value` prop is being "copied" to the state
- Changes in the dialog then trigger changes in `state.value`
- The `props.value` remain unchanged until `handleSaveAndClose` is called

It sort of makes sense to work like this, because it also needs to be possible to click cancel on the dialog and in that case the `props.value` will still be correct.

HOWEVER... Now comes the weird part... Before this fix, the `props` (not the `state`) were being spread onto the `ExpressionDialog`. And the result of this was that whenever the dropdown value changed, it seemed to be ignoring this and just rendered the old value from the props. So I've changed this part. My changes will also provide values from the state to the ExpressionManager.... Even though the ExpressionManager was working fine before, I still think this change is for the better. 

I suspect that the ExpressionManager has some form of internal state and treats its `descriptionValue` and `formulaValue` as "initial values". That would certainly explain why this was working fine before even though we were actually providing it with bad values...

Reply:
It does seem very complex (not your changes, the existing logic) for a rather simple UI pattern. With a custom store implementation, and forking props to state, etc. Though you could argue that in this case forking props to state is warranted (see the second tweet): 

<img width="663" alt="Screenshot 2019-12-02 at 14 26 32" src="https://user-images.githubusercontent.com/7355199/69963179-ec37e400-150f-11ea-8067-bad8f0d7394d.png">
(https://twitter.com/dan_abramov/status/979520339968516097?lang=en)

In general improving this seems like it'd warrant way more in-depth structural changes. You changes make sense to me, and function how I'd expect them to, so they like a decent solution to me.

Reply:
I agree, it's in general an antipattern to copy props to state, and its done several places in the app, probably from before this was considered an antipattern. This might be a case where it's actually fine though. And as long as it works I see no reason to use time and change it now.